src/communication/communicator.hpp

```diff
@@ -64,30 +66,37 @@ class communicator {
         //   -> src_domains: array with one entry for every local connection
         // Also the count of presynaptic sources from each domain
         //   -> src_counts: array with one entry for each domain
+
+        // Record all the gid in a flat vector.
+        // These are used to map from local index to gid in the parallel loop
+        // that populates gid_infos.
+        std::vector<cell_gid_type> gids;
+        gids.reserve(num_local_cells_);
+        for (auto g: dom_dec.groups) {
+            util::append(gids, g.gids);
+        }
+        // Build the connection information for local cells in parallel.
         std::vector<gid_info> gid_infos;
-        gid_infos.reserve(dom_dec.num_local_cells);
+        gid_infos.resize(num_local_cells_);
+        threading::parallel_for::apply(0, gids.size(),
+            [&](cell_size_type i) {
+                auto gid = gids[i];
+                gid_infos[i] = gid_info(gid, i, rec.connections_on(gid));
+            });
 
-        cell_local_size_type n_cons = 0;
-        cell_local_size_type n_gid = 0;
+        cell_local_size_type n_cons =
+            util::sum_by(gid_infos, [](const gid_info& g){ return g.conns.size(); }, 0);
```

Comment:
Integer typing here is a bit obscure. Is the `0` added to `sum_by` to force an `int` return value, that then gets promoted to the the unsigned `cell_local_size_type`?

I think it would be clearer to leave out the '0', and let the conversion from `size_t` to `cell_local_size_type` be implicit.

Reply:
Good point, will change!